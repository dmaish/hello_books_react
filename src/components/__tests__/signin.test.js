import React from "react";
import { shallow } from "enzyme";
import SignIn from "../signin";

describe("My signIn component", () => {
	it("should render correctly in \"debug\" mode", () => {
		const wrapper = shallow(<SignIn/>)
        expect(wrapper).toMatchSnapshot()
    })
})

//make our assertion that form in component is being rendered
// make sure the inputs on the signin form also render correctly
describe("Sigin component renders well", () => {
	it("form renders", () => {
		expect(shallow(<SignIn/>).find("form.signInForm").exists()).toBe(true)
	}),
	it("email field renders", () => {
		expect(shallow(<SignIn/>).find("#loginEmail").length).toEqual(1)
	}),
	it("password field renders", () => {
		expect(shallow(<SignIn/>).find("#loginPassword").length).toEqual(1)
	})
})

