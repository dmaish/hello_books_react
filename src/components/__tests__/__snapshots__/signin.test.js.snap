// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My signIn component should render correctly in "debug" mode 1`] = `
<div
  className="container container-signin"
>
  <div
    className="card"
  >
    <div
      className="card-header"
    >
      <h5>
        signin
      </h5>
    </div>
    <div
      className="card-body"
    >
      <form
        className="signInForm"
        onSubmit={[Function]}
      >
        <div
          className="form-group row"
        >
          <div
            className="col-sm-2"
          />
          <label
            className="col-sm-2 col-form-label "
          >
            email
          </label>
          <div
            className="col-sm-6 "
          >
            <input
              className="form-control "
              name="email"
              onChange={[Function]}
              placeholder="email"
              required={true}
              type="email"
              value=""
            />
          </div>
          <div
            className="col-sm-2"
          />
        </div>
        <div
          className="form-group row"
        >
          <div
            className="col-sm-2"
          />
          <label
            className="col-sm-2 col-form-label "
          >
            password
          </label>
          <div
            className="col-sm-6 "
          >
            <input
              className="form-control "
              data-minlength="8"
              name="password"
              onChange={[Function]}
              placeholder="password"
              required={true}
              type="password"
              value=""
            />
          </div>
          <div
            className="col-sm-2"
          />
        </div>
        <div
          className="form-group row"
        >
          <div
            className="col-sm-4"
          />
          <div
            className="col-sm-4"
          >
            <button
              className="btn btn-primary"
              type="submit"
            >
              submit
            </button>
          </div>
          <div
            className="col-sm-4"
          />
        </div>
      </form>
    </div>
  </div>
</div>
`;
